kolibri/core/assets/src/core-actions.js

```diff
@@ -349,6 +443,85 @@ function stopTrackingProgress(store, Kolibri) {
   updateTimeSpent(store, Kolibri, true);
 }
 
+function saveMasteryLog(store, Kolibri) {
+  const masteryLogModel = Kolibri.resources.MasteryLog.getModel(store.core.logging.mastery.id);
+  masteryLogModel.save(_masteryLogModel(store)).then((newMasteryLog) => {
```

Comment:
It would probably be good to add an additional check for the attemptLog, where if the save completes after we have moved onto a new attempt, we should not set this. For the masteryLog, the change should be on page, yes.